File: JAVA/ElectronicStore.java
// class produk elektronik
public class ElectronicStore {
    // atribut yang bersifat rahasia
    private String productID;
    private String category;
    private String brand;
    private String type;
    private long price;
    private int stock;
    private int warrantyYears;

    // constructor untuk inisialisasi atribut
    public ElectronicStore(String id, String category, String brand, String type, long price, int stock, int wy) {
        this.productID = id;
        this.category = category;
        this.brand = brand;
        this.type = type;
        this.price = price;
        this.stock = stock;
        this.warrantyYears = wy;
    }

    // getter untuk mengakses atribut
    public String getProductID() {
        return productID;
    }

    // setter untuk mengubah atribut
    public void setCategory(String category) {
        this.category = category;
    }
    public void setBrand(String brand) {
        this.brand = brand;
    }
    public void setType(String type) {
        this.type = type;
    }
    public void setPrice(long price) {
        this.price = price;
    }
    public void setStock(int stock) {
        this.stock = stock;
    }

    // method untuk menampilkan semua detail produk dengan rapi
    public void displayProductDetails() {
        System.out.println(" ID        : " + productID);
        System.out.println(" Category  : " + category);
        System.out.println(" Brand     : " + brand);
        System.out.println(" Type      : " + type);
        System.out.println(" Price     : Rp " + price);
        System.out.println(" Stock     : " + stock + " units");
        System.out.println(" Warranty  : " + warrantyYears + " years");
    }
}

File: CPP/ElectronicStore.cpp
#include <iostream>
#include <string>

using namespace std;

class ElectronicStore {
// atribut yang bersifat rahasia
private:
    string productID;
    string category;   
    string brand;      
    string type;       
    long price;
    int stock;
    int warrantyYears;

public: 
    // constructor untuk inisialisasi atribut
    ElectronicStore(string id = "", string category = "", string brand = "", string type = "", long price = 0, int stock = 0, int wy=0){
        this->productID = id;
        this->category = category;
        this->brand = brand;
        this->type = type;
        this->price = price;
        this->stock = stock;
        this->warrantyYears = wy;
    }

    // getter untuk mengakses atribut
    string getProductID(){
        return productID;
    }

    // setter untuk mengubah atribut
    void setCategory(string category){
        this->category = category;
    }
    void setBrand(string brand){
        this->brand = brand;
    }
    void setType(string type){ 
        this->type = type; 
    }
    void setPrice(long price){
        this->price = price;
    }
    void setStock(int stock){
        this->stock = stock;
    }
   
    // method untuk menampilkan semua detail produk dengan rapi
    void displayProductDetails(){
        cout << " ID            : " << productID << endl;
        cout << " Category      : " << category << endl;
        cout << " Brand         : " << brand << endl;
        cout << " Type          : " << type << endl;
        cout << " Price         : Rp " << price << endl;
        cout << " Stock         : " << stock << " units" << endl;
        cout << " Warranty      : " << warrantyYears << " years" << endl;
    } 
};
